ykrt/src/location.rs

```diff
@@ -334,9 +321,25 @@ impl Drop for Location {
         let ls = self.load(Ordering::Relaxed);
         if !ls.is_counting() {
             debug_assert!(!ls.is_locked());
-            let hr = unsafe { ls.hot_location() };
-            unsafe {
-                Box::from_raw(hr);
+            self.lock().unwrap();
```

Comment:
swap lines 324 and 325 to reduce the critical section slightly?

Reply:
or do you require an absolutely up-to-date `hl` for correctness?

Reply:
That would be incorrect: you can't deref the `hot_location` without locking.

Reply:
Can the deref and locking operations be combined to prevent programming errors? i.e. doing a deref does an implicit lock? I guess that become annoying when you have different modes of locking (lock and try_lock).

Maybe I talked myself out of it.

Reply:
> Can the deref and locking operations be combined to prevent programming errors?

Not unless you know something that I don't ;)